### PR TITLE
Failed to reset value in retained_map on erase

### PR DESCRIPTION
**Pull request by kleunen**

### CODE REVIEW

File: test/retained_topic_map.hpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 #include <boost/optional.hpp>
 #include <boost/range/adaptor/reversed.hpp>
+#include <boost/none.hpp>
```
Comment (redboltz) [inline test/retained_topic_map.hpp:18]: Is this needed?

File: test/retained_topic_map.hpp [hunk 2]
```diff
@@ -197,6 +198,10 @@ class retained_topic_map {
     bool erase_topic(MQTT_NS::string_view topic) {
         auto path = find_topic(topic);
 
+        // Reset the value
+        if(!path.empty())
+            path.back()->second.value = MQTT_NS::optional<Value>();
```
Comment (redboltz) [inline test/retained_topic_map.hpp:203]:
```suggestion
            path.back()->second.value = MQTT_NS::nullopt;
```

It's not mandatory but more comprehensible.

See https://github.com/redboltz/mqtt_cpp/blob/master/include/mqtt/optional.hpp#L35

Comment (kleunen) [reply test/retained_topic_map.hpp:203]: Yes, it is annoying you have to name the type to reset the value for MQTT_NS::optional<Value>()

**Comment by kleunen** [comment]
There is still an issue in this PR, so don't merge yet

**Comment by kleunen** [comment]
Now should work OK

**Comment by redboltz** [comment]
LGTM, thanks!

**Pull request merged**